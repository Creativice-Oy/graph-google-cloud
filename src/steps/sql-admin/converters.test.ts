import { sqladmin_v1beta4 } from 'googleapis';
import {
  createCloudMySQLInstanceEntity,
  createCloudPostgresInstanceEntity,
  createCloudSQLServerInstanceEntity,
} from './converters';

function getMockSQLInstance(
  partial?: Partial<sqladmin_v1beta4.Schema$DatabaseInstance>,
): sqladmin_v1beta4.Schema$DatabaseInstance {
  return {
    kind: 'sql#instance',
    state: 'RUNNABLE',
    settings: {
      authorizedGaeApplications: [],
      tier: 'db-custom-1-3840',
      kind: 'sql#settings',
      availabilityType: 'ZONAL',
      pricingPlan: 'PER_USE',
      activationPolicy: 'ALWAYS',
      ipConfiguration: {
        authorizedNetworks: [],
        ipv4Enabled: true,
      },
      locationPreference: {
        zone: 'europe-west3-b',
        kind: 'sql#locationPreference',
      },
      dataDiskType: 'PD_SSD',
      maintenanceWindow: {
        kind: 'sql#maintenanceWindow',
        hour: 0,
        day: 0,
      },
      backupConfiguration: {
        startTime: '06:00',
        kind: 'sql#backupConfiguration',
        location: 'eu',
        backupRetentionSettings: {
          retentionUnit: 'COUNT',
          retainedBackups: 7,
        },
        enabled: true,
        replicationLogArchivingEnabled: true,
        pointInTimeRecoveryEnabled: true,
        transactionLogRetentionDays: 7,
      },
      settingsVersion: '19',
      storageAutoResizeLimit: '0',
      storageAutoResize: true,
      dataDiskSizeGb: '10',
    },
    etag: '98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d',
    ipAddresses: [],
    instanceType: 'CLOUD_SQL_INSTANCE',
    project: 'j1-gc-integration-dev-300716',
    serviceAccountEmailAddress:
      'p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com',
    selfLink:
      'https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres',
    connectionName:
      'j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres',
    name: 'cloud-sql-postgres',
    region: 'europe-west3',
    gceZone: 'europe-west3-b',
    ...partial,
  };
}

describe('#createCloudMySQLInstanceEntity', () => {
  test('should convert to entity', () => {
    expect(
      createCloudMySQLInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'MYSQL_5_7',
          backendType: 'SECOND_GEN',
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should convert to entity with localInfile param set to off', () => {
    expect(
      createCloudMySQLInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'MYSQL_5_7',
          backendType: 'SECOND_GEN',
          settings: {
            databaseFlags: [
              {
                name: 'local_infile',
                value: 'off',
              },
            ],
          },
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should convert to entity with requireSSL param set to true', () => {
    expect(
      createCloudMySQLInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'MYSQL_5_7',
          backendType: 'SECOND_GEN',
          settings: {
            ipConfiguration: {
              requireSsl: true,
            },
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createCloudPostgresInstanceEntity', () => {
  test('should convert to entity', () => {
    expect(
      createCloudPostgresInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'POSTGRES_12',
          backendType: 'SECOND_GEN',
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should convert to entity with logCheckpoints param set to off', () => {
    expect(
      createCloudPostgresInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'POSTGRES_12',
          backendType: 'SECOND_GEN',
          settings: {
            databaseFlags: [
              {
                name: 'log_checkpoints',
                value: 'off',
              },
            ],
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createCloudSQLServerInstanceEntity', () => {
  test('should convert to entity', () => {
    expect(
      createCloudSQLServerInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'SQLSERVER_2017_STANDARD',
          backendType: 'SECOND_GEN',
        }),
      ),
    ).toMatchSnapshot();
  });

  test('should convert to entity with crossDbOwnershipChaining param set to on', () => {
    expect(
      createCloudSQLServerInstanceEntity(
        getMockSQLInstance({
          databaseVersion: 'SQLSERVER_2017_STANDARD',
          backendType: 'SECOND_GEN',
          settings: {
            databaseFlags: [
              {
                name: 'cross db ownership chaining',
                value: 'on',
              },
            ],
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});
