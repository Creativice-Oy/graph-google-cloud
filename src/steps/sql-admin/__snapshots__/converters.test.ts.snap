// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createCloudMySQLInstanceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "MYSQL_5_7",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "activationPolicy": "ALWAYS",
          "authorizedGaeApplications": Array [],
          "availabilityType": "ZONAL",
          "backupConfiguration": Object {
            "backupRetentionSettings": Object {
              "retainedBackups": 7,
              "retentionUnit": "COUNT",
            },
            "enabled": true,
            "kind": "sql#backupConfiguration",
            "location": "eu",
            "pointInTimeRecoveryEnabled": true,
            "replicationLogArchivingEnabled": true,
            "startTime": "06:00",
            "transactionLogRetentionDays": 7,
          },
          "dataDiskSizeGb": "10",
          "dataDiskType": "PD_SSD",
          "ipConfiguration": Object {
            "authorizedNetworks": Array [],
            "ipv4Enabled": true,
          },
          "kind": "sql#settings",
          "locationPreference": Object {
            "kind": "sql#locationPreference",
            "zone": "europe-west3-b",
          },
          "maintenanceWindow": Object {
            "day": 0,
            "hour": 0,
            "kind": "sql#maintenanceWindow",
          },
          "pricingPlan": "PER_USE",
          "settingsVersion": "19",
          "storageAutoResize": true,
          "storageAutoResizeLimit": "0",
          "tier": "db-custom-1-3840",
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_mysql_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "createdOn": undefined,
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "localInfile": "on",
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "MySQL",
}
`;

exports[`#createCloudMySQLInstanceEntity should convert to entity with localInfile param set to off 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "MYSQL_5_7",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "databaseFlags": Array [
            Object {
              "name": "local_infile",
              "value": "off",
            },
          ],
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_mysql_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "createdOn": undefined,
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "localInfile": "off",
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "MySQL",
}
`;

exports[`#createCloudMySQLInstanceEntity should convert to entity with requireSSL param set to true 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "MYSQL_5_7",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "ipConfiguration": Object {
            "requireSsl": true,
          },
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_mysql_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "createdOn": undefined,
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "localInfile": "on",
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "name": "cloud-sql-postgres",
  "requireSSL": true,
  "type": "MySQL",
}
`;

exports[`#createCloudPostgresInstanceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "POSTGRES_12",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "activationPolicy": "ALWAYS",
          "authorizedGaeApplications": Array [],
          "availabilityType": "ZONAL",
          "backupConfiguration": Object {
            "backupRetentionSettings": Object {
              "retainedBackups": 7,
              "retentionUnit": "COUNT",
            },
            "enabled": true,
            "kind": "sql#backupConfiguration",
            "location": "eu",
            "pointInTimeRecoveryEnabled": true,
            "replicationLogArchivingEnabled": true,
            "startTime": "06:00",
            "transactionLogRetentionDays": 7,
          },
          "dataDiskSizeGb": "10",
          "dataDiskType": "PD_SSD",
          "ipConfiguration": Object {
            "authorizedNetworks": Array [],
            "ipv4Enabled": true,
          },
          "kind": "sql#settings",
          "locationPreference": Object {
            "kind": "sql#locationPreference",
            "zone": "europe-west3-b",
          },
          "maintenanceWindow": Object {
            "day": 0,
            "hour": 0,
            "kind": "sql#maintenanceWindow",
          },
          "pricingPlan": "PER_USE",
          "settingsVersion": "19",
          "storageAutoResize": true,
          "storageAutoResizeLimit": "0",
          "tier": "db-custom-1-3840",
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_postgres_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "createdOn": undefined,
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "logCheckpoints": "off",
  "logConnections": "off",
  "logDisconnections": "off",
  "logLockWaits": "off",
  "logMinDurationStatement": "-1",
  "logMinErrorStatement": "error",
  "logTempFiles": "0",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "Postgres",
}
`;

exports[`#createCloudPostgresInstanceEntity should convert to entity with logCheckpoints param set to off 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "POSTGRES_12",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "databaseFlags": Array [
            Object {
              "name": "log_checkpoints",
              "value": "off",
            },
          ],
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_postgres_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "createdOn": undefined,
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "logCheckpoints": "off",
  "logConnections": "off",
  "logDisconnections": "off",
  "logLockWaits": "off",
  "logMinDurationStatement": "-1",
  "logMinErrorStatement": "error",
  "logTempFiles": "0",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "Postgres",
}
`;

exports[`#createCloudSQLServerInstanceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "SQLSERVER_2017_STANDARD",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "activationPolicy": "ALWAYS",
          "authorizedGaeApplications": Array [],
          "availabilityType": "ZONAL",
          "backupConfiguration": Object {
            "backupRetentionSettings": Object {
              "retainedBackups": 7,
              "retentionUnit": "COUNT",
            },
            "enabled": true,
            "kind": "sql#backupConfiguration",
            "location": "eu",
            "pointInTimeRecoveryEnabled": true,
            "replicationLogArchivingEnabled": true,
            "startTime": "06:00",
            "transactionLogRetentionDays": 7,
          },
          "dataDiskSizeGb": "10",
          "dataDiskType": "PD_SSD",
          "ipConfiguration": Object {
            "authorizedNetworks": Array [],
            "ipv4Enabled": true,
          },
          "kind": "sql#settings",
          "locationPreference": Object {
            "kind": "sql#locationPreference",
            "zone": "europe-west3-b",
          },
          "maintenanceWindow": Object {
            "day": 0,
            "hour": 0,
            "kind": "sql#maintenanceWindow",
          },
          "pricingPlan": "PER_USE",
          "settingsVersion": "19",
          "storageAutoResize": true,
          "storageAutoResizeLimit": "0",
          "tier": "db-custom-1-3840",
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_sql_server_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "containedDatabaseAuthentication": "off",
  "createdOn": undefined,
  "crossDatabaseOwnershipChaining": "off",
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "SQLServer",
}
`;

exports[`#createCloudSQLServerInstanceEntity should convert to entity with crossDbOwnershipChaining param set to on 1`] = `
Object {
  "_class": Array [
    "Database",
  ],
  "_key": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "backendType": "SECOND_GEN",
        "connectionName": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
        "databaseVersion": "SQLSERVER_2017_STANDARD",
        "etag": "98de3deaeeca4420e21a891488c187b0b029235e2e0cb551fc00a388d1ac4f7d",
        "gceZone": "europe-west3-b",
        "instanceType": "CLOUD_SQL_INSTANCE",
        "ipAddresses": Array [],
        "kind": "sql#instance",
        "name": "cloud-sql-postgres",
        "project": "j1-gc-integration-dev-300716",
        "region": "europe-west3",
        "selfLink": "https://www.googleapis.com/sql/v1beta4/projects/j1-gc-integration-dev-300716/instances/cloud-sql-postgres",
        "serviceAccountEmailAddress": "p165882964161-x5ifig@gcp-sa-cloud-sql.iam.gserviceaccount.com",
        "settings": Object {
          "databaseFlags": Array [
            Object {
              "name": "cross db ownership chaining",
              "value": "on",
            },
          ],
        },
        "state": "RUNNABLE",
      },
    },
  ],
  "_type": "google_cloud_sql_sql_server_instance",
  "authorizedNetworks": Array [],
  "automatedBackupsEnabled": true,
  "containedDatabaseAuthentication": "off",
  "createdOn": undefined,
  "crossDatabaseOwnershipChaining": "on",
  "displayName": "cloud-sql-postgres",
  "hasPublicIP": false,
  "location": "j1-gc-integration-dev-300716:europe-west3:cloud-sql-postgres",
  "name": "cloud-sql-postgres",
  "requireSSL": false,
  "type": "SQLServer",
}
`;
